infra/Pos/Reporting/Methods.hs

```diff
@@ -183,6 +178,11 @@ withCompressedLogs files action = do
         curTime <- formatTime defaultTimeLocale "%q" <$> getCurrentTime
         tempDir <- getTemporaryDirectory
         pure $ tempDir <//> ("report-" <> take 6 curTime <> ".tar.lzma")
+    compressStrict = BSL.toStrict .
+                     compressWith
+                     defaultCompressParams
+                     { compressLevel = CompressionLevel0 } .
+                     BSL.fromStrict
```

Comment:
Please stop converting to and from lazy and strict bytestrings. Think about what is going on. If this is all done properly with lazy bytestrings then you actually get proper streaming behaviour and it'll all work in O(1) space.

So: don't use `BSL.toStrict` on the result of `Tar.write`, don't use it on the result of `compressWith`. Write the file using `BSL.writeFile`.

Also, why do we need to do a custom tar packing rather than using `Tar.pack`? The latter is doing the streaming thing right so that we don't need to open lots of file `Handle`s at once.

Reply:
>Also, why do we need to do a custom tar packing rather than using Tar.pack? 

There is a subtle difference. `tarPackIndependently` takes a list of _absolute_ paths, but packs them as they are in the same directory. For example, i have `/home/user/a.log` and `/var/lib/something.log`, but i don't want `tar` to extract it using full paths. I'm not sure why this useful feature wasn't implemented in `tar` itself, so I've added it here.

Also, what's a profit? Reporting server limit is 20mb, it doesn't really matter if we're doing streaming or not, performance bottleneck is IO/packing anyway.

@DanielRrr please continue the discussion!

Reply:
@dcoutts 
>Please stop converting to and from lazy and strict bytestrings.

Fixed.